CMake/resolve_dependency_modules/fmt.cmake

```diff
@@ -13,21 +13,20 @@
 # limitations under the License.
 include_guard(GLOBAL)
 
-if(DEFINED ENV{VELOX_FMT_URL})
-  set(VELOX_FMT_SOURCE_URL "$ENV{VELOX_FMT_URL}")
-else()
-  set(VELOX_FMT_VERSION 8.0.1)
-  set(VELOX_FMT_SOURCE_URL
-      "https://github.com/fmtlib/fmt/archive/${VELOX_FMT_VERSION}.tar.gz")
-  set(VELOX_FMT_BUILD_SHA256_CHECKSUM
-      b06ca3130158c625848f3fb7418f235155a4d389b2abc3a6245fb01cb0eb1e01)
-endif()
+set(VELOX_FMT_VERSION 8.0.1)
+set(VELOX_FMT_BUILD_SHA256_CHECKSUM
+    b06ca3130158c625848f3fb7418f235155a4d389b2abc3a6245fb01cb0eb1e01)
+set(VELOX_FMT_SOURCE_URL
+    "https://github.com/fmtlib/fmt/archive/${VELOX_FMT_VERSION}.tar.gz")
+set_with_default(VELOX_FMT_SOURCE_URL VELOX_FMT_URL ${VELOX_FMT_SOURCE_URL})
+set_with_default(VELOX_FMT_BUILD_SHA256_CHECKSUM VELOX_FMT_SHA256
```

Comment:
By pulling the sha definition into the variable an empty hash no longer causes an error. This makes it possible to omit the sha when providing a custom url but leave the option open to provide a sha for supply chain security if desired by the user. :+1:

Reply:
But actually you are also kind of right @kgpai , the users has to set the `VELOX_FMT_SHA256` envvar either way either with a sha matching the custom file or an empty string. Otherwise `set_with_default` will set it to the default sha which will probably fail (maybe not if it's just for an offline build). I don't really like having to set the empty string... 

@Yohahaha could you wrap the two calls to `set_with_default` in a macro and pass in the package name as an arg and guarding the second call/unsetting the sha in case a custom url is set but no sha? (a macro seems best here as otherwise you will have to forward all vars into `PARENT_SCOPE`)